packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -266,11 +407,40 @@ class GrpcRouteSpec extends RouteSpec {
   public readonly match: GrpcRouteMatch;
   public readonly timeout?: GrpcTimeout;
 
+  /**
+   * The retry policy.
+   */
+  public readonly retryPolicy?: GrpcRetryPolicy;
+
   constructor(props: GrpcRouteSpecOptions) {
     super();
     this.weightedTargets = props.weightedTargets;
     this.match = props.match;
     this.timeout = props.timeout;
+
+    if (props.retryPolicy) {
+      if (props.retryPolicy.grpcRetryEvents && props.retryPolicy.grpcRetryEvents.length === 0) {
+        throw new Error('Specify at least one value in `grpcRetryEvents` or leave it undefined');
+      }
+
+      if (props.retryPolicy.httpRetryEvents && props.retryPolicy.httpRetryEvents.length === 0) {
+        throw new Error('Specify at least one value in `httpRetryEvents` or leave it undefined');
+      }
+
+      if (props.retryPolicy.tcpRetryEvents && props.retryPolicy.tcpRetryEvents.length === 0) {
+        throw new Error('Specify at least one value in `tcpRetryEvents` or leave it undefined');
+      }
```

Comment:
Same here (get rid of this validation, and return `undefined` for an empty list in `renderGrpcRetryPolicy `).

Reply:
Ok. No problem.